openscm/adapters/dice.py

```diff
@@ -0,0 +1,388 @@
+
+    _views: Any
+    """Parameter views"""
+
+    _year2100: int
+    """Timestamp of 2100-01-01"""
+
+    def _initialize_model(self) -> None:
+        """
+        Initialize the model.
+        """
+        self._period_length = YEAR
+        self._year2100 = int(datetime(2100, 1, 1).timestamp())  # TODO 365 days-year
+
+        parameter_names = list(MODEL_PARAMETER_DEFAULTS.keys()) + [
+            "E",
+            "M_atm",
+            "M_l",
+            "M_u",
+            "T_atm",
+            "T_ocean",
+            "force",
+            "b11",
+            "b21",
+            "b22",
+            "b32",
+            "b33",
+        ]
+        self._values = namedtuple("DICEValues", parameter_names)
+        self._views = namedtuple("DICEViews", parameter_names)
+
+        for name, (default, unit) in MODEL_PARAMETER_DEFAULTS.items():
+            setattr(self._values, name, default)
+            self._parameters.get_writable_scalar_view(
+                ("DICE", name), ("World",), unit
+            ).set(default)
+            setattr(
+                self._views,
+                name,
+                self._parameters.get_scalar_view(("DICE", name), ("World",), unit),
+            )
+
+    def _initialize_model_input(self) -> None:
+        pass
+
+    def _initialize_run_parameters(self) -> None:
+        self._timestep = 0
+        self._timestep_count = (
+            self._stop_time - self._start_time
+        ) // self._period_length
+
+        time_points = create_time_points(
+            self._start_time,
+            self._period_length,
+            self._timestep_count,
+            ParameterType.POINT_TIMESERIES,
+        )
+        time_points_for_averages = create_time_points(
+            self._start_time,
+            self._period_length,
+            self._timestep_count,
+            ParameterType.AVERAGE_TIMESERIES,
+        )
+
+        # Total CO2 emissions (GtCO2 per year)
+        self._views.E = self._parameters.get_timeseries_view(
+            ("Emissions", "CO2"),
+            ("World",),
+            "GtCO2/a",
+            time_points_for_averages,
+            ParameterType.AVERAGE_TIMESERIES,
+            InterpolationType.LINEAR,
+            ExtrapolationType.LINEAR,
+        )
+
+        # TODO remove setting missing emissions once there is a method in tests for that
+        if self._views.E.is_empty:
+            self._parameters.get_writable_timeseries_view(
+                ("Emissions", "CO2"),
+                ("World",),
+                "GtCO2/a",
+                time_points_for_averages,
+                ParameterType.AVERAGE_TIMESERIES,
+                InterpolationType.LINEAR,
+                ExtrapolationType.LINEAR,
+            ).set(np.zeros(self._timestep_count))
+
+        # Concentration in atmosphere
+        self._views.M_atm = self._output.get_writable_timeseries_view(
+            ("Concentration", "Atmosphere"),
+            ("World",),
+            "GtC",
```

Comment:
following above comments, this unit makes no sense for atmospheric concentrations (you can't convert GtC into ppm with our current unit_registry) so I think we have to rename this variable

Reply:
as said above these are the original names (you should also the economic variables) and i will change them once we have a standard